sycl/doc/extensions/USM/USM.adoc

```diff
@@ -222,56 +222,88 @@ While the modern C++ `usm_allocator` interface is sufficient for specifying USM
  * `const sycl::device& dev` - the SYCL `device` to allocate on
  * `const sycl::context& ctxt` - the SYCL `context` to which `device` belongs
 
 Return value:: Returns a pointer to the newly allocated memory on the specified `device` on success.   Memory allocated by `sycl::malloc_device` must be deallocated with `sycl::free` to avoid memory leaks. On failure, returns `nullptr`.
 
 [source,cpp]
 ----
+(1)
 void* sycl::malloc_device(size_t size,
                           const sycl::queue& q);
+
+(2)
+template <typename T>
+T* sycl::malloc_device(size_t count,
+                        const sycl::queue& q);
 ----
 
 Parameters::
- * `size_t size` - number of bytes to allocate
+ * (1) `size_t size` - number of bytes to allocate
+ * (2) `size_t count` - number of elements of type `T` to allocate
  * `const sycl::queue& q` - the SYCL `q` that provides the `device` and `context` to allocate against
 
 Return value:: Returns a pointer to the newly allocated memory on the `device` associated with `q` on success.   Memory allocated by `sycl::malloc_device` must be deallocated with `sycl::free` to avoid memory leaks. On failure, returns `nullptr`.
 
 ===== aligned_alloc
 [source,cpp]
 ----
+(1)
 void* sycl::aligned_alloc_device(size_t alignment,
                                  size_t size,
                                  const sycl::device& dev, 
                                  const sycl::context& ctxt);
+
+(2)
+template <typename T>
+T* sycl::aligned_alloc_device(size_t alignment,
+                              size_t count,
+                              const sycl::device& dev, 
+                              const sycl::context& ctxt);
 ----
 
 Parameters::
   * `size_t alignment` - specifies the byte alignment.  Must be a valid alignment supported by the implementation.
-  * `size_t size` - number of bytes to allocate
+  * (1) `size_t size` - number of bytes to allocate
+  * (2) `size_t count` - number of elements of type `T` to allocate
   * `const sycl::device& dev` - the `device` to allocate on
   * `const sycl::context& ctxt` - the SYCL `context` to which `device` belongs
 Return value:: Returns a pointer to the newly allocated memory on the specified `device` on success.  Memory allocated by `sycl::aligned_alloc_device` must be deallocated with `sycl::free` to avoid memory leaks.  On failure, returns `nullptr`.
 
 [source,cpp]
 ----
+(1)
 void* sycl::aligned_alloc_device(size_t alignment,
                                  size_t size,
                                  const sycl::queue& q);
+
+(2)
+template <typename T>
+T* sycl::aligned_alloc_device(size_t alignment,
+                              size_t count,
+                              const sycl::queue& q);
 ----
 
 Parameters::
   * `size_t alignment` - specifies the byte alignment.  Must be a valid alignment supported by the implementation.
-  * `size_t size` - number of bytes to allocate
+  * (1) `size_t size` - number of bytes to allocate
```

Comment:
Should we encode the units in the name?  `size_bytes`?

Reply:
If you prefer.